app/src/main/java/io/appium/uiautomator2/server/mjpeg/MjpegScreenshotStream.java

```diff
@@ -119,8 +118,15 @@ private void matchFramerate(long targetInterval, long start) {
         }
     }
 
+    private static synchronized UiAutomation getUiAutomation() {
+        if (UI_AUTOMATION == null) {
```

Comment:
caching must not be applied there. Please change this method body to `return CustomUiDevice.getInstance().getUiAutomation()`

Reply:
Done, removed the method in https://github.com/appium/appium-uiautomator2-server/pull/463/commits/c2be84093da4ea9e8c9138c277664d000f60574f